plugins/inputs/zfs/zfs_freebsd.go

```diff
@@ -87,6 +87,47 @@ func (z *Zfs) gatherPoolStats(acc telegraf.Accumulator) (string, error) {
 	return strings.Join(pools, "::"), nil
 }
 
+func (z *Zfs) gatherDatasetStats(acc telegraf.Accumulator) (string, error) {
+	properties := []string{"name", "avail", "used", "usedsnap", "usedds"}
+
+	lines, err := z.zdataset(properties)
+	if err != nil {
+		return "", err
+	}
+
+	datasets := []string{}
+	for _, line := range lines {
+		col := strings.Split(line, "\t")
+
+		datasets = append(datasets, col[0])
+	}
+
+	if z.DatasetMetrics {
+		for _, line := range lines {
+			col := strings.Split(line, "\t")
+			if len(col) != len(properties) {
+				z.Log.Warnf("Invalid number of columns for line: %s", line)
+				continue
+			}
+
+			tags := map[string]string{"dataset": col[0]}
+			fields := map[string]interface{}{}
+
+			for i, key := range properties[1:] {
+				value, err := strconv.ParseInt(col[i+1], 10, 64)
+				if err != nil {
+					return "", fmt.Errorf("Error parsing %s: %s", key, err)
```

Comment:
parseInt uses col[i+1] but the error uses key. maybe change them both to key? can also use %q to quote strings

Reply:
@ssoroka key and col are not the same. Key is for column name and col is an array containing the values for 1 line of result.

Reply:
@ssoroka I don't understand want you want to do?

Reply:
Ok in that case I'd like to change the error message to include what was difficult to parse. eg:
```suggestion
				value, err := strconv.ParseInt(col[i+1], 10, 64)
				if err != nil {
					return "", fmt.Errorf("Error parsing %s %q: %s", key, col[i+1], err)
```